client/components/data/query-tld-validation-schema/index.jsx

```diff
@@ -0,0 +1,52 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { difference, isEmpty, isEqual, keys } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getValidationSchemas } from 'state/selectors';
+import { requestValidationSchemas } from 'state/domains/management/validation-schemas/actions';
+
+export class QueryTldValidationSchema extends Component {
+	static propTypes = {
+		validationSchemas: PropTypes.object.isRequired,
+		requestValidationSchemas: PropTypes.func.isRequired,
+		tlds: PropTypes.array.isRequired,
+	};
+
+	componentWillMount() {
+		this.fetchMissingSchemas( this.props.tlds );
```

Comment:
I believe the general advice is to make calls to remote endpoints in `componentDidMount`. `componentWillMount` is also executed on server side rendering, therefore 2 x. 

https://reactjs.org/docs/react-component.html#mounting-componentdidmount